packages/compiler-cli/src/ngtsc/annotations/component/src/handler.ts

```diff
@@ -1222,14 +1222,25 @@ export class ComponentDecoratorHandler
       // Register all Directives and Pipes used at the top level (outside
       // of any defer blocks), which would be eagerly referenced.
       const eagerlyUsed = new Set<ClassDeclaration>();
-      for (const dir of bound.getEagerlyUsedDirectives()) {
-        eagerlyUsed.add(dir.ref.node);
-      }
-      for (const name of bound.getEagerlyUsedPipes()) {
-        if (!pipes.has(name)) {
-          continue;
+
+      if (this.enableHmr) {
+        // In HMR we need to preserve all the dependencies, because they have to remain consitent
```

Comment:
nit: spelling of comment

```suggestion
        // In HMR we need to preserve all the dependencies, because they have to remain consistent
```